main.cpp

```diff
@@ -483,6 +581,8 @@ struct Memory
     int limitRamConsumption(double limitAmount, int totalAppsAllowed = 3);
     // overclock RAM
     bool overclockRam(bool overheated = false);
+    // new member function
+    Memory newMemoryComponent(int threshold, int startingValue);
```

Comment:
a Memory instance will return another Memory?
that doesn't make much sense.
This is a member function, so the only way to call it is by using a Memory instance...
- [ ] revise